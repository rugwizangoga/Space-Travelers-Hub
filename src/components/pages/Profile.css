.missionsList li {
    border: 1px solid black;
    padding: 10px;
    list-style: none;
    width: 50%;
    margin-left: -3%;
  }
  
  h2 {
    margin-bottom: 12px;
  }
  
  .myprofile__list {
    border: 1px solid black;
  }