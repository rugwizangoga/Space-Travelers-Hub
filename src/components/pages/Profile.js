import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { selectAllMissions } from '../../redux/missions/missionsSlice';

const MissionItem = ({ missionName }) => <li>{missionName}</li>;
MissionItem.propTypes = {
  missionName: PropTypes.string.isRequired,
};

const Profile = () => {
  const missions = useSelector(selectAllMissions);
  const joinedMissions = missions.filter((mission) => mission.joined === true);
  const rocketsList = useSelector((state) => state.rockets.rockets);
  const reservedRockets = rocketsList.filter((rocket) => rocket.reserved);
  return (
    <div className="profile-container">
      <section>
        <h2>My Missions</h2>
        <ul className="missionsList">
          {joinedMissions.map((mission) => (
            <MissionItem
              key={mission.mission_id}
              missionName={mission.mission_name}
            />
          ))}
        </ul>
      </section>
      <section>
        <h2>My Rockets</h2>
        <ul className="missionsList">
          {reservedRockets.map((rocket) => (
            <MissionItem
              key={rocket.id}
              missionName={rocket.name}
            />
          ))}
        </ul>
      </section>
    </div>
  );
};

export default Profile;
