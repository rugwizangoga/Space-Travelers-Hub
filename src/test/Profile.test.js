import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../components/pages/Profile';
import store from '../redux/configureStore';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

describe('Missions component', () => {
  it('should render correctly', () => {
    const MyProfile = TestRenderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(MyProfile).toMatchSnapshot();
  });
});
