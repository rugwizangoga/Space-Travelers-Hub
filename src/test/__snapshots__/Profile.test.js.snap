// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component should render correctly 1`] = `
<div
  className="profile-container"
>
  <section>
    <h2>
      My Missions
    </h2>
    <ul
      className="missionsList"
    />
  </section>
  <section>
    <h2>
      My Rockets
    </h2>
    <ul
      className="missionsList"
    />
  </section>
</div>
`;
